lms/djangoapps/courseware/views/views.py

```diff
@@ -741,7 +741,9 @@ def _progress(request, course_key, student_id):
         'student': student,
         'passed': is_course_passed(course, grade_summary),
         'credit_course_requirements': _credit_course_requirements(course_key, student),
-        'certificate_data': _get_cert_data(student, course, course_key, is_active, enrollment_mode)
+        'certificate_data': _get_cert_data(student, course, course_key, is_active, enrollment_mode),
+        'time_zone': request.user.preferences.model.get_value(request.user, "time_zone", None),
+        'language': request.user.preferences.model.get_value(request.user, "pref-lang", None)
```

Comment:
To prevent the 2 extra sql roundtrips to the database, add `preferences` to the `prefetch_related` call above on [L720](https://github.com/edx/edx-platform/pull/13794/files#diff-2749747f3edc5443c734c6a6546e91d1R720).